driver/gpio/imxgpio/imx6ull.hpp

```diff
@@ -0,0 +1,577 @@
+    { IMX_MAKE_PIN_1(4, 26), 0x484, 0x000010B0, 0x1F8, 0x05 }, // GPIO4_IO26 CSI_DATA05
+    { IMX_MAKE_PIN_1(4, 27), 0x488, 0x000010B0, 0x1FC, 0x05 }, // GPIO4_IO27 CSI_DATA06
+    { IMX_MAKE_PIN_1(4, 28), 0x48C, 0x000010B0, 0x200, 0x05 }, // GPIO4_IO28 CSI_DATA07
+  };
+
+  // i.MX6 SOC specific Select Input/Alt Function/Pin map
+  //
+  // iMX6 Dual/Quad
+  const IMX_PIN_INPUT_DATA Imx6ULLGpioPinInputSelectTable[] = {
+    //--------------------------------------------------
+    // Pin Absolute Number | Pin Alt Mode value | Select Input Reg Offset | Input Select value
+    //--------------------------------------------------
+    { IMX_MAKE_PIN_1(1, 0), 0, 0x5AC, 1 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO00 - ALT0 - I2C2_SCL - IOMUXC_I2C2_SCL_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 0), 1, 0x58C, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO00 - ALT1 - GPT1_CAPTURE1 - IOMUXC_GPT1_CAPTURE1_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 0), 2, 0x4B8, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO00 - ALT2 - ANATOP_OTG1_ID - IOMUXC_USB_OTG1_ID_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 0), 3, 0x574, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO00 - ALT3 - ENET1_REF_CLK1 - IOMUXC_ENET1_REF_CLK1_SELECT_INPUT
+
+    { IMX_MAKE_PIN_1(1, 1), 0, 0x5B0, 1 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO01 - ALT0 - I2C2_SDA - IOMUXC_I2C2_SDA_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 1), 2, 0x664, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO01 - ALT2 - USB_OTG1_OC - IOMUXC_USB_OTG_OC_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 1), 3, 0x57C, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO01 - ALT3 - ENET2_REF_CLK2 - IOMUXC_ENET2_REF_CLK2_SELECT_INPUT
+
+    { IMX_MAKE_PIN_1(1, 2), 0, 0x5A4, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO02 - ALT0 - I2C1_SCL - IOMUXC_I2C1_SCL_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 2), 4, 0x66C, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO02 - ALT4 - USDHC1_WP - IOMUXC_USDHC1_WP_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 2), 6, 0x610, 1 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO02 - ALT6 - SDMA_EXT_EVENT00 - IOMUXC_SDMA_EVENTS0_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 2), 8, 0x624, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO02 - ALT8 - UART1_TX - IOMUXC_UART1_RX_DATA_SELECT_INPUT
+
+    { IMX_MAKE_PIN_1(1, 3), 0, 0x5A8, 1 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO03 - ALT0 - I2C1_SDA - IOMUXC_I2C1_SDA_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 3), 2, 0x660, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO03 - ALT2 - USB_OTG2_OC - IOMUXC_USB_OTG2_OC_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 3), 4, 0x668, 0 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO03 - ALT4 - USDHC1_CD_B - IOMUXC_USDHC1_CD_B_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 3), 8, 0x624, 1 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO03 - ALT8 - UART1_RX - IOMUXC_UART1_RX_DATA_SELECT_INPUT
+
+    { IMX_MAKE_PIN_1(1, 4), 0, 0x574, 1 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO04 - ALT0 - ENET1_REF_CLK1 - IOMUXC_ENET1_REF_CLK1_SELECT_INPUT
+    { IMX_MAKE_PIN_1(1, 4), 8, 0x644, 2 }, // IOMUXC_SW_MUX_CTL_PAD_GPIO1_IO04 - ALT8 - UART5_TX - IOMUXC_UART5_RX_DATA_SELECT_INPUT

```

Comment:
We don't support reversing the send/receive pins (DCE/DTE mode) in the UART driver, so any entries that map UARTx_TX to UARTx_RX and UARTx_RX to UARTx_TX should be commented out.